Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStore.mm

```diff
@@ -578,14 +578,15 @@ static bool hasValue(const ImageBufferBackendHandle& backendHandle)
 
     layer.contentsOpaque = m_isOpaque;
 #if HAVE(SUPPORT_HDR_DISPLAY_APIS)
-    layer.contentsHeadroom = m_maxRequestedEDRHeadroom;
     ALLOW_DEPRECATED_DECLARATIONS_BEGIN
     if (layer.wantsExtendedDynamicRangeContent) {
         // Painted contents have already been tonemapped, so disable it on the layer.
         if (isDelegatedDisplay)
             layer.toneMapMode = CAToneMapModeIfSupported;
-        else
+        else {
             layer.toneMapMode = CAToneMapModeNever;
+            layer.contentsHeadroom = m_maxRequestedEDRHeadroom;
```

Comment:
This clash between contentsHeadroom and CAToneMapModeIfSupported doesn't seem to be documented (but I trust you've tested it). Should this be reported to the appropriate people, so they fix it or at least document it?